RequistionsMgt/requistions/models.py

```diff
@@ -1,133 +1,133 @@
     p_flag=models.BooleanField(default=True)
-    objects = models.Manager()
 
     class Meta:
-        verbose_name_plural='periods'
+        verbose_name_plural='Periods'
     def __str__ (self):
         return self.period_name
 
-class employee(models.Model):
+class Employee(models.Model):
     emp_name=models.CharField(max_length=200,verbose_name='employee name')
-    staff_id_num=models.CharField(max_length=20,unique=True,verbose_name='UNRAID No.')
-    emp_station=models.ForeignKey(station,on_delete=models.CASCADE)
-    emp_position=models.ForeignKey(position,on_delete=models.CASCADE)
-    emp_scale=models.ForeignKey(salaryscale,on_delete=models.CASCADE)
+    staff_id_num=models.CharField(max_length=20, unique=True, verbose_name='UNRAID No.')
+    emp_station=models.ForeignKey(Station, on_delete=models.CASCADE)
+    emp_position=models.ForeignKey(Position, on_delete=models.CASCADE)
+    emp_scale=models.ForeignKey(SalaryScale, on_delete=models.CASCADE)
     date_left_org=models.DateField(null=True)
-    objects = models.Manager()
     class Meta:
-        verbose_name_plural='employees'
+        verbose_name_plural='Employees'
     def __str__ (self):
         return self.emp_name
 
-class requistiondetailtype(models.Model):
+class RequisitionDetailType(models.Model):
     ## e.g request for NOA,SDA,Advance,expense claim
     detail_name=models.CharField(max_length=100)
     objects = models.Manager()
     class Meta:
-        verbose_name_plural='requistiondetailtypes'
+        verbose_name_plural='Requisition detail types'
     def __str__ (self):
         return self.detail_name
-class requistion(models.Model):
-    made_by=models.ForeignKey(employee,on_delete=models.CASCADE)
+class Requisition(models.Model):
+    made_by=models.ForeignKey(Employee, on_delete=models.CASCADE)
     date_made=models.DateTimeField(auto_now_add=True)
     purpose=models.TextField(verbose_name='Purpose')
-    objects = models.Manager()
+    total =models.DecimalField(max_digits=10, decimal_places=2, default=0.0)
     class Meta:
-        verbose_name_plural='requistion'
+        verbose_name_plural='Requisition'
     def __str__ (self):
         return self.purpose
-class requistiondetail(models.Model):
-    requistionid=models.ForeignKey(requistion,on_delete=models.CASCADE)
-    emp_name=models.ForeignKey(employee,on_delete=models.CASCADE)
-    detail_type=models.ForeignKey(requistiondetailtype,on_delete=models.CASCADE)
-    period_claimed=models.ForeignKey(period,on_delete=models.CASCADE)
-    ndays=models.IntegerField(default=0,verbose_name="Number of days")
-    startdate=models.DateField(null=True)
-    enddate=models.DateField(null=True)
-    rate=models.FloatField(default=0.0,verbose_name='Allowances Rate')
-    amount=models.FloatField(default=0.0,verbose_name='Amount')
-    objects = models.Manager()
+class RequisitionDetail(models.Model):
+    requisition = models.ForeignKey(Requisition, null=True, on_delete=models.CASCADE)
```

Comment:
This was the issue, I only added
```python
null=True
``` 
and re-ran `python manage.py makemigrations` the `migrate`  that fixed the issue.

Reply:
this should explain the error and also give you more understanding on the issue https://www.youtube.com/watch?v=pMHtofYy_L0